app/code/Magento/BundleGraphQl/Model/Resolver/BundleItemLinks.php

```diff
@@ -24,39 +23,25 @@ class BundleItemLinks implements ResolverInterface
      */
     private $linkCollection;
 
-    /**
-     * @var ValueFactory
-     */
-    private $valueFactory;
-
     /**
      * @param Collection $linkCollection
-     * @param ValueFactory $valueFactory
      */
     public function __construct(
-        Collection $linkCollection,
-        ValueFactory $valueFactory
+        Collection $linkCollection
     ) {
         $this->linkCollection = $linkCollection;
-        $this->valueFactory = $valueFactory;
     }
 
     /**
-     * {@inheritDoc}
+     * @inheritdoc
      */
-    public function resolve(Field $field, $context, ResolveInfo $info, array $value = null, array $args = null) : Value
+    public function resolve(Field $field, $context, ResolveInfo $info, array $value = null, array $args = null)
     {
         if (!isset($value['option_id']) || !isset($value['parent_id'])) {
-            $result = function () {
-                return null;
-            };
-            return $this->valueFactory->create($result);
+            throw new GraphQlInputException(__('"option_id" and "parent_id" values should be specified'));
         }
-        $this->linkCollection->addIdFilters((int)$value['option_id'], (int)$value['parent_id']);
-        $result = function () use ($value) {
-            return $this->linkCollection->getLinksForOptionId((int)$value['option_id']);
-        };
 
-        return $this->valueFactory->create($result);
+        $this->linkCollection->addIdFilters((int)$value['option_id'], (int)$value['parent_id']);
```

Comment:
This was not supposed to be replaced because this was actual performance optimization via Deferred mechanism.

Reply:
Fixed